src/Mod/Part/App/TopoShapeOpCode.h

```diff
@@ -0,0 +1,99 @@
+/****************************************************************************
+ *   Copyright (c) 2018 Zheng Lei (realthunder) <realthunder.dev@gmail.com> *
+ *                                                                          *
+ *   This file is part of the FreeCAD CAx development system.               *
+ *                                                                          *
+ *   This library is free software; you can redistribute it and/or          *
+ *   modify it under the terms of the GNU Library General Public            *
+ *   License as published by the Free Software Foundation; either           *
+ *   version 2 of the License, or (at your option) any later version.       *
+ *                                                                          *
+ *   This library  is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of         *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
+ *   GNU Library General Public License for more details.                   *
+ *                                                                          *
+ *   You should have received a copy of the GNU Library General Public      *
+ *   License along with this library; see the file COPYING.LIB. If not,     *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,          *
+ *   Suite 330, Boston, MA  02111-1307, USA                                 *
+ *                                                                          *
+ ****************************************************************************/
+
+#ifndef PART_TOPOSHAPE_OPCODE_H
+#define PART_TOPOSHAPE_OPCODE_H
+
+/** Definition of commonly used TopoShape operational code
+ *
+ * The operational code (op code) is encoded into the mapped element name to
+ * provide more meaning shape history when tracking back. Some code can be
+ * passed to generalized shape making API TopoShape::makEBoolean() to make
+ * shape.
```

Comment:
Some slight grammatical strangeness here. How about something like:
```suggestion
 * The operational code (op code) is encoded into the mapped element name to
 * provide more context and meaning to the  shape history when tracing it back
 * to its predecessors. Some op codes can be passed to the generalized shape-
 * making API TopoShape::makEBoolean() to make a shape.
```
Also: why do all these TopoShape::makE functions have a capital "E" in their function name? is this some kind of convention from OCC?

Reply:
Probably @realthunder  would be better confirming if the change removes some semantics.

My best guess about `makE` is that it stands for `make E...` where E stands for something.

Reply:
`E` stand for `Element`. I choose the naming to distinguish the element mapping aware interface from others in the transition period. Once it gets matured, we can drop that convention.